frontend/controller/actions/group.js

```diff
@@ -309,27 +342,28 @@ export default (sbp('sbp/selectors/register', {
           data: CSKp,
           // The OP_ACTION_ENCRYPTED is necessary to let the DM counterparty
           // that a chatroom has just been created
-          permissions: [GIMessage.OP_KEY_SHARE, GIMessage.OP_ACTION_ENCRYPTED],
-          allowedActions: ['gi.contracts/identity/joinDirectMessage'],
+          permissions: [GIMessage.OP_ACTION_ENCRYPTED + '#inner'],
+          allowedActions: ['gi.contracts/identity/joinDirectMessage#inner'],
           purpose: ['sig'],
           ringLevel: Number.MAX_SAFE_INTEGER,
           name: `${contractID}/${CSKid}`
         }],
-        signingKeyId: findKeyIdByName(rootState[userID], 'csk')
+        signingKeyId: sbp('chelonia/contract/currentKeyIdByName', userID, 'csk')
       })
 
       // Share our PEK with the group so that group members can see
       // our name and profile information
-      const PEKid = findKeyIdByName(rootState[userID], 'pek')
-      const revokedPEKids = findRevokedKeyIdsByName(rootState[userID], 'pek')
+      const keyIds = sbp('chelonia/contract/historicalKeyIdsByName', userID, 'pek')
 
-      PEKid && await sbp('gi.actions/out/shareVolatileKeys', {
+      keyIds.length && await sbp('gi.actions/out/shareVolatileKeys', {
         contractID: contractID,
         contractName: 'gi.contracts/group',
         subjectContractID: userID,
-        keyIds: [PEKid, ...revokedPEKids]
+        keyIds: keyIds
       })
 
+      */
+
```

Comment:
If all this code is no longer needed should it be deleted?

Reply:
This is related to how the group creator also joins normally (and it can be deleted).